### PR TITLE
[DependencyInjection] Sort services in service locator according to priority

### LINKED ISSUE #42506: [DependencyInjection] Priority ignored for tagged locator
**Symfony version(s) affected**: 5.3.6

**Description**
Currently it's allowed to configure priority on `!tagged_locator` but it's ignored while dumping container.

**How to reproduce**
```yaml
App\Handler\HandlerLocator:
    arguments:
        $locator: !tagged_locator
            tag: app.handler
            index_by: 'id'
            default_index_method: 'getId'
            default_priority_method: 'getPriority'
```

```php
class HandlerLocator
{
    public function __construct(private ServiceLocator $locator)
    {
    }

    public function get(string $id): HandlerInterface
    {
        return $this->locator->get($id);
    }

    public function available(): array
    {
        return array_keys($this->locator->getProvidedServices());
    }
}
```

```php
// Outputs with wrong order
foreach ($this->locator->available() as $id) {
    $output->writeln($id);
}
```

**Additional context**
Priority does not make much sense in locator but I have a form type which contains all handlers and I want to output them according to priority

### PR DESCRIPTION
**Pull request by BoShurik**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #42506
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Compiler/ServiceLocatorTagPass.php [hunk 1]
```diff
@@ -102,7 +102,6 @@ public static function register(ContainerBuilder $container, array $refMap, stri
             }
             $refMap[$id] = new ServiceClosureArgument($ref);
         }
-        ksort($refMap);
```
Comment (BoShurik) [inline src/Symfony/Component/DependencyInjection/Compiler/ServiceLocatorTagPass.php:105]: As I understand sort here is for easy testing

**Comment by nicolas-grekas** [comment]
Can you please add a test case that covers the new behavior?

**Comment by nicolas-grekas** [comment]
Oh, and please have a look at failing test cases.

**Comment by nicolas-grekas** [comment]
Friendly ping @BoShurik

**Comment by BoShurik** [comment]
@nicolas-grekas yes, it's in my todo-list for this week

**Comment by BoShurik** [comment]
@nicolas-grekas I am not sure how to fix `7.4, high-deps` tests.
As I understand in this case my changes in `DependencyInjection` used as a dependency for `HttpKernel` tests and those tests falls
So we need to exclude `symfony/dependency-injection: 5.4` for `symfony/http-kernel: 4.4`. Is it possible?

**Comment by nicolas-grekas** [comment]
>  So we need to exclude symfony/dependency-injection: 5.4 for symfony/http-kernel: 4.4. Is it possible?

no need to exclude: we should patch 4.4 instead to relax the test case, which is too strict apparently

**Review by nicolas-grekas — APPROVED** [review]
(but please send a PR for 4.4 to relax the tests)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @BoShurik.

**Pull request closed without merging**